napalm/ios/ios.py

```diff
@@ -1955,6 +1955,54 @@ def get_snmp_information(self):
             snmp_dict['chassis_id'] = snmp_chassis
         return snmp_dict
 
+    def get_users(self):
+        """
+        Returns a dictionary with the configured users.
+        The keys of the main dictionary represents the username.
+        The values represent the details of the user,
+        represented by the following keys:
+
+            * level (int)
+            * password (str)
+            * sshkeys (list)
+
+        *Note: sshkeys on ios is the ssh key fingerprint
+
+        The level is an integer between 0 and 15, where 0 is the
+        lowest access and 15 represents full access to the device.
+        """
+        users = {}
+        command = "show run | section username"
+        output = self._send_command(command)
+        curuser = ""
+        for line in output.splitlines():
+            if line.startswith("username"):
+                tokens = line.split(" ")
+                users[tokens[1]] = {
+                    'level': 1,
+                    'password': "",
+                    'sshkeys': []
+                }
+                i = 2
+                while i < len(tokens):
```

Comment:
I think this parsing will probably work better with regular expressions...I am happy to help write this, but basically cover the patterns of `privilege` and `secret` using regex.

This shifting and incrementing I think is harder to read and easier to make a mistake on.

Reply:
I agree. Do you have a good example to point me at?

Reply:
Here is a first pass at it:

Where first username is:

```
username pyclass privilege 15 secret 5 $1$CQCe$poonxxxzzzttabct/3eXub41
```

```
>>> re.search(r"^username\s+(?P<username>\S+)\s+privilege\s+(?P<priv_level>\S+)\s+secret \d+\s+(?P<pwd_hash>\S+)$", output, re.M).groupdict()
{'username': 'pyclass', 'priv_level': '15', 'pwd_hash': '$1$CQCe$poonxxxzzzttabct/3eXub41'}
```

Reply:
We would need to account for a few scenarios i.e. when privilege is there or not (if it is password and not secret, et cetera).